diff --git a/build.gradle b/build.gradle
index 79f3e6284..b191c1cd5 100644
--- a/build.gradle
+++ b/build.gradle
@@ -258,6 +258,10 @@
     runtimeOnly("mekanism:Mekanism:${mekanism_version}")
     runtimeOnly("mekanism:Mekanism:${mekanism_version}:generators")
 
+    // Create: Crafts & Additions
+    compileOnly("maven.modrinth:createaddition:${createaddition_version}")
+    runtimeOnly("maven.modrinth:createaddition:${createaddition_version}")
+
     // JEI
     compileOnly("mezz.jei:${jeiSlug}-neoforge-api:${jei_version}")
     devRuntimeOnly("mezz.jei:${jeiSlug}-neoforge:${jei_version}")
diff --git a/gradle.properties b/gradle.properties
index d4b7851c0..fe103c8d8 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -25,6 +25,7 @@ ae2_version=19.2.17
 cct_version=1.120.0
 jei_version=19.27.0.336
 mekanism_version=1.21.1-10.7.19.85
+createaddition_version=neoforge-1.21.1-1.6.0
 modernfix_version=5.27.11+mc1.21.1
 projred_version=1.21.1-4.22.0
 ccl_version=1.21.1-4.6.1.529
diff --git a/src/main/scala/li/cil/oc/integration/Mods.scala b/src/main/scala/li/cil/oc/integration/Mods.scala
index caa6c60b7..1eaf5f513 100644
--- a/src/main/scala/li/cil/oc/integration/Mods.scala
+++ b/src/main/scala/li/cil/oc/integration/Mods.scala
@@ -26,6 +26,7 @@ object Mods {
   val JustEnoughItems = new SimpleMod(IDs.JustEnoughItems)
   val AppliedEnergistics2 = new SimpleMod(IDs.AppliedEnergistics2)
   val Mekanism = new SimpleMod(IDs.Mekanism)
+  val CreateAddition = new SimpleMod(IDs.CreateAddition)
   val Minecraft = new SimpleMod(IDs.Minecraft)
   val OpenComputers = new SimpleMod(IDs.OpenComputers)
   val EnderStorage = new SimpleMod(IDs.EnderStorage)
@@ -40,6 +41,7 @@ object Mods {
     integration.minecraft.ModMinecraft,
     integration.computercraft.ModComputerCraft,
     integration.appeng.ModAppliedEnergistics2.INSTANCE,
+    integration.createaddition.ModCreateAddition.INSTANCE,
     integration.enderstorage.ModEnderStorage,
     integration.projectred.ModProjectRed,
 
@@ -101,6 +103,7 @@ object Mods {
     final val JustEnoughItems = "jei"
     final val AppliedEnergistics2 = "ae2"
     final val Mekanism = "mekanism"
+    final val CreateAddition = "createaddition"
     final val Minecraft = "minecraft"
     final val OpenComputers = "opencomputers"
     final val EnderStorage = "enderstorage"
